### PR TITLE
Add test TestSigBroacastTwoPerSig

### PR DESCRIPTION
**Pull request by algonautshant**
## Summary

Test the broadcast function to make sure each locally generated signature is broadcast twice per period, and each received message is broadcast once.

### CODE REVIEW

File: stateproof/worker_test.go [hunk 1]
```diff
@@ -108,31 +110,33 @@ func (s *testWorkerStubs) addBlock(ccNextRound basics.Round) {
...
-			signerInRound := part.StateProofSecrets.GetSigner(uint64(rnd))
-			partRecordForRound := account.StateProofRecordForRound{
-				ParticipationRecord: partRecord,
-				StateProofSecrets:   signerInRound,
-			}
-			out = append(out, partRecordForRound)
+	return stateProofRecordFromKeys(rnd, s.keys)
+}
+
+func stateProofRecordFromKeys(rnd basics.Round, keys []account.Participation) (out []account.StateProofRecordForRound) {
```
Comment (algonathan) [inline stateproof/worker_test.go:116]: Why have you created this function? You feed it with the same keys as you have in the `testWorkerStubs` in the test below. why not call `s.stateProofRecordFromKeys(rnd)` instead?

File: stateproof/worker_test.go [hunk 2]
```diff
@@ -546,3 +550,104 @@ func TestSignerDoesntDeleteKeysWhenDBDoesntStoreSigs(t *testing.T) {
...
+		spw.broadcastSigs(basics.Round(brnd), proto)
+	}
+
+	close(tns.sigmsg)
+	wg.Wait()
+
+	// Verify the number of times each signature was broadcast
+	for addr, sb := range signatureBcasted {
+		if fromThisNode[addr] {
+			require.Equal(t, 2*periods, sb)
```
Comment (algonathan) [inline stateproof/worker_test.go:648]: I'm not certain this test makes certain each interval has exactly two broadcasts per address that are from the node.
can you change it to be a more granular test? That is, check that per period 2 signatures are present, and not an aggregation of all period results?

**Review by algonathan — COMMENTED** [review]
Overall, the test is well-written, but i did add a few questions.

File: stateproof/worker_test.go (hunk 1 above)
Comment (algonautshant) [reply stateproof/worker_test.go:116]: Because I do not have the object `s` `testWorkerStubs`.

**Review by algoidan — APPROVED** [review]
LGTM

**Pull request merged**